### PR TITLE
Do not inject String.class without explicit binging

### LINKED ISSUE #4: Version 1.1.0 of guice-extension breaks parameterized tests.
After updating Guice-Extension to the new version, the parameterized tests broke down.
```java
@ExtendWith(GuiceExtension.class)
public class Test {
    @ParameterizedTest
    @ValueSource(strings = {"one", "two", "three"})
    void test(String count) {
        assertTrue(true);
    }
}
```

Error:
```java
org.junit.jupiter.api.extension.ParameterResolutionException: Discovered multiple competing ParameterResolvers for parameter [java.lang.String arg0] in method [void com.tests.Test.test(java.lang.String)]: name.falgout.jeffrey.testing.junit.guice.GuiceExtension@19e4653c, org.junit.jupiter.params.ParameterizedTestParameterResolver@795509d9
```

I researched this problem and found that the reason is built-in binding for String.class.
String has empty constructor and binds in runtime by Guice to "" (empty string).

See line 206:
https://github.com/JeffreyFalgout/junit5-extensions/commit/adf2f4ece977e834ba9b4f1563ebc568784ad3a4#diff-5eef6748419687f2efdf650d6649f118

I suggest to add additional annotation with bindingModeParam(ALL,EXPLICIT_ONLY) to set binding mode and process it in `supportsParameter` method.

..or just make workaround with built-in binding String.class.

I would like to hear your opinion and suggestions on this issue.

### PR DESCRIPTION
**Pull request by SergeyAlekseevN**
PR to resolve #4 issue.
+ update deps.

### CODE REVIEW

File: guice-extension/src/test/java/name/falgout/jeffrey/testing/junit/guice/ParametrizedTestCompatibilityTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        @ParameterizedTest
+        @ValueSource(strings = "valueSourceString")
+        void parametrizedTestWithStringsShouldWorkWithGuiceExtension(String value) {
+            assertEquals("valueSourceString", value);
+        }
+
+        @ExtendWith(GuiceExtension.class)
+        @ParameterizedTest
+        @ValueSource(classes = Integer.class)
+        void parametrizedTestWithStringsShouldWorkWithGuiceExtension(Class<Object> clazz) {
```
Comment (JeffreyFalgout) [inline guice-extension/src/test/java/name/falgout/jeffrey/testing/junit/guice/ParametrizedTestCompatibilityTest.java]:
```suggestion
        void parametrizedTestWithoutStringsShouldWorkWithGuiceExtension(Class<Object> clazz) {
```

Comment (JeffreyFalgout) [reply guice-extension/src/test/java/name/falgout/jeffrey/testing/junit/guice/ParametrizedTestCompatibilityTest.java]: I think this is what you meant?

File: guice-extension/src/main/java/name/falgout/jeffrey/testing/junit/guice/GuiceExtension.java [hunk 2]
```diff
@@ -220,6 +220,11 @@ public boolean supportsParameter(ParameterContext parameterContext,
         parameter);
     Optional<Injector> optInjector = getInjectorForParameterResolution(extensionContext);
     return optInjector.filter(injector -> {
+
+      // Do not bind String without explicit bindings.
+      if (key.equals(Key.get(String.class)) && injector.getExistingBinding(key) == null)
```
Comment (JeffreyFalgout) [inline guice-extension/src/main/java/name/falgout/jeffrey/testing/junit/guice/GuiceExtension.java]: Add { } to this if

File: class-argument-converter/pom.xml [hunk 3]
```diff
@@ -40,6 +40,12 @@
       <artifactId>junit-jupiter-params</artifactId>
       <scope>compile</scope>
     </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
```
Comment (JeffreyFalgout) [inline class-argument-converter/pom.xml]: Why do all of these poms need this dependency all of a sudden?

File: guice-extension/pom.xml [hunk 4]
```diff
@@ -39,10 +39,20 @@
       <groupId>org.junit.platform</groupId>
       <artifactId>junit-platform-commons</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-params</artifactId>
```
Comment (JeffreyFalgout) [inline guice-extension/pom.xml]:
```suggestion
      <artifactId>junit-jupiter-params</artifactId>
      <scope>test</scope>
```

File: guice-extension/src/test/java/name/falgout/jeffrey/testing/junit/guice/ParametrizedTestCompatibilityTest.java (hunk 1 above)
Comment (SergeyAlekseevN) [reply guice-extension/src/test/java/name/falgout/jeffrey/testing/junit/guice/ParametrizedTestCompatibilityTest.java]: yes.

File: class-argument-converter/pom.xml (hunk 3 above)
Comment (SergeyAlekseevN) [reply class-argument-converter/pom.xml]: > Why do all of these poms need this dependency all of a sudden?

I updated surefire-plugin and removed junit-platform-surefire-provider.

See:
https://junit.org/junit5/docs/current/user-guide/#running-tests-build-maven-engines-configure
Starting with version 2.22.0, Maven Surefire and Maven Failsafe provide native support for executing tests on the JUnit Platform. The pom.xml file in the junit5-jupiter-starter-maven project demonstrates how to use the Maven Surefire plugin and can serve as a starting point for configuring your Maven build.

In order to have Maven Surefire or Maven Failsafe run any tests at all, at least one TestEngine implementation must be added to the test classpath.

**Comment by SergeyAlekseevN** [comment]
I fixed the code.

**Comment by JeffreyFalgout** [comment]
This will be available as part of [1.1.1](https://github.com/JeffreyFalgout/junit5-extensions/releases/tag/junit5-extensions-1.1.1).

**Pull request merged**